common/stats/stats.h

```diff
@@ -83,6 +88,11 @@ class Stats {
   void AddMetric(const uint32_t metric, int64_t value);
   void AddMetric(const std::string& metric, int64_t value);
 
+  // Register callback for Counter / Metric so they will be reported
+  // periodically by the Stats class.
+  void RegisterIncr(const std::string& counter, std::function<uint64_t()>);
+  void RegisterAddMetric(const std::string& metric, std::function<int64_t()>);
```

Comment:
we may provide a uint32_t version counter and metric APIs if we need to use it now. Otherwise, we may add it later.

Reply:
the stats function is uint64_t() for ThriftServer